process/block/interceptedBlocks/interceptedBlockHeader.go

```diff
@@ -86,8 +91,47 @@ func (inHdr *InterceptedHeader) CheckValidity() error {
 	return inHdr.hdr.CheckChainID(inHdr.chainID)
 }
 
+func (inHdr *InterceptedHeader) isEpochCorrect() bool {
+	if inHdr.shardCoordinator.SelfId() != sharding.MetachainShardId {
+		return true
+	}
+	if inHdr.epochStartTrigger.EpochStartRound() >= inHdr.epochStartTrigger.EpochFinalityAttestingRound() {
+		return true
+	}
+	if inHdr.hdr.GetEpoch() >= inHdr.epochStartTrigger.Epoch() {
+		return true
+	}
+	if inHdr.hdr.GetRound() <= inHdr.epochStartTrigger.EpochStartRound() {
+		return true
+	}
+	if inHdr.hdr.GetRound() <= inHdr.epochStartTrigger.EpochFinalityAttestingRound()+process.EpochChangeGracePeriod {
+		return true
+	}
+
+	return false
+}
+
 // integrity checks the integrity of the header block wrapper
 func (inHdr *InterceptedHeader) integrity() error {
+	if !inHdr.isEpochCorrect() {
+		return fmt.Errorf("%w : shard header with old epoch and bad round: "+
+			"round=%v, "+
+			"shardEpoch=%v, "+
+			"shardHeaderHash=%s, "+
+			"shardId=%v, "+
+			"metaEpoch=%v, "+
+			"shardRound=%v, "+
+			"metaFinalityAttestingRound=%v ",
+			process.ErrEpochDoesNotMatch,
+			inHdr.hdr.Round,
```

Comment:
This should be the chronology round. This value is printed again in line 131.

Reply:
yes! good catch, it was duplicate. Removed the round because there is no chronology here.